File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsPlatformCoursesAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class PlatformCourseRow(
    val id: String,
    val courseCode: String,
    val title: String,
    val status: String,
    val orgId: String,
    val orgName: String,
    val instructorName: String? = null,
    val termId: String? = null,
    val termName: String? = null,
    val enrollmentCount: Long = 0,
    val createdAt: String = "",
    val updatedAt: String = "",
)

@Serializable
data class PaginatedPlatformCourses(
    val items: List<PlatformCourseRow> = emptyList(),
    val total: Long = 0,
    val page: Int = 1,
    val perPage: Int = 25,
    val totalPages: Int = 0,
)

@Serializable
data class CoursesDashboardStats(
    val createdLast7Days: Long = 0,
    val activeCourses: Long = 0,
    val draftCourses: Long = 0,
    val totalCourses: Long = 0,
    val archivedCourses: Long = 0,
)

enum class CoursesListFilter(val apiValue: String) {
    Created7d("created_7d"),
    Active("active"),
    Draft("draft"),
    Total("total"),
    Archived("archived"),
}

@Serializable
data class PlatformCourseReport(
    val id: String,
    val courseCode: String,
    val title: String,
    val description: String? = null,
    val status: String = "",
    val orgId: String = "",
    val orgName: String = "",
    val instructorName: String? = null,
    val termId: String? = null,
    val termName: String? = null,
    val enrollmentCount: Long = 0,
    val published: Boolean = false,
    val archived: Boolean = false,
    val createdAt: String = "",
    val updatedAt: String = "",
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsPlatformCoursesAdmin.swift
import Foundation

struct PlatformCourseRow: Decodable, Identifiable, Hashable {
    var id: String
    var courseCode: String
    var title: String
    var status: String
    var orgId: String
    var orgName: String
    var instructorName: String?
    var termId: String?
    var termName: String?
    var enrollmentCount: Int64
    var createdAt: String
    var updatedAt: String
}

struct PaginatedPlatformCourses: Decodable {
    var items: [PlatformCourseRow]
    var total: Int64
    var page: Int
    var perPage: Int
    var totalPages: Int
}

struct CoursesDashboardStats: Decodable, Equatable {
    var createdLast7Days: Int64
    var activeCourses: Int64
    var draftCourses: Int64
    var totalCourses: Int64
    var archivedCourses: Int64
}

enum CoursesListFilter: String, CaseIterable, Identifiable, Hashable {
    case created7d = "created_7d"
    case active
    case draft
    case total
    case archived
    var id: String { rawValue }
}

struct PlatformCourseReport: Decodable {
    var id: String
    var courseCode: String
    var title: String
    var description: String?
    var status: String
    var orgId: String
    var orgName: String
    var instructorName: String?
    var termId: String?
    var termName: String?
    var enrollmentCount: Int64
    var published: Bool
    var archived: Bool
    var createdAt: String
    var updatedAt: String
}
